### PR TITLE
Brings Back Buckshot (Voidshot)

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
(Props to @necromanceranne for sprites)

[Recently got the greenlight from the headcoder triumvirate to bring back buckshot 🎉🎉🎉](https://i.imgur.com/y7v6I9u.png)

I don't want to just bring it back as how it used to be though, where a shotgun full of buckshot let you 2-tap anyone to death at point blank in half a second like a stunbaton on massive steroids. Like with rubbershot in #64020, I want crew shotgun usage to have their own niche, hopefully leaning towards maint tunnels, since that's about as CQB-friendly of an environment as you can have in SS13. I freestyled these main changes below as ideas for how to do that without just putting them back on top of the mountain like before:

- Regular (military) buckshot remains in the hands of the Syndies, with a few small changes. It deals a bit more damage (7.5 -> 9 dmg/pellet, 6*9=54 brute), wounds a bit better, and can now ricochet once (making it better in maint tunnels).
- Adds Voidshot, a new shotgun shell type orderable by cargo for 250 credits for a box of 7 (requires armory access to open). Compared to buckshot, Voidshot has one less pellet, deals a bit less damage (5*8=40 brute), and is weak against armor, but has a tighter spread, better range, and much better wounding power. Doesn't have the same killing power as buckshot, but can definitely wound (and thus debuff) the hell out of any soft targets. Can also ricochet twice, making it even better in maint tunnels.
- Buckshot/voidshot/rubbershot now all slow down somewhat after the first few tiles traveled, encouraging up-close use. Rubbershot slows down more than the others, buckshot is quicker up close, voidshot retains its speed better at range.

These are all spitball ideas, I really wanted to throw this up to get feedback/ideas. Now that the crew has the weaker, more debuff-y voidshot for itself, we can make buckshot for syndies/military/antags a bit deadlier and more satisfying, while still giving sec/the crew ways to maim each other that doesn't scale as well against armored enemies. Also I like the idea of pushing shotguns towards maint tunnels is cool. Leave your thoughts below though!

## Why It's Good For The Game
The triumphant return of crew CQB firepower, while keeping Syndies/military troops as more outright lethal. Adds a bit more variety to the different kinds of guns and places they excel in, further pushing shotguns as effective in close-quarters maint tunnels.

## Changelog
[these changes aren't final yet]

### CODE REVIEW

**Comment by optimumtact** [comment]
Will always be the best combat choice for crew because every space on ss13 is a tight space

shortsighted decision here

**Comment by wesoda25** [comment]
> Will always be the best combat choice for crew because every space on ss13 is a tight space
> 
> shortsighted decision here that just reduces the value of every other gun available to the crew

Not if we re-add hitscans.

**Comment by SinfulBliss** [comment]
> Will always be the best combat choice for crew because every space on ss13 is a tight space
> 
> shortsighted decision here that just reduces the value of every other gun available to the crew

This isn't true. Even normal buckshot wasn't very effective over a few tiles away because of the massive spread, including in tight spaces since it didn't have ricochet. 

Looks like it'll finally give shotguns some usage aside from setting all your allies on fire with incendiaries. What's the damage falloff with armor like for the void shells? Does it still have a wound chance?

**Comment by Ryll-Ryll** [comment]
> Will always be the best combat choice for crew because every space on ss13 is a tight space
> 
> shortsighted decision here that just reduces the value of every other gun available to the crew

Most gun combat happens further than 3 tiles away, which is where you're guaranteed to miss 1 or 2 pellets at least. Outside of direct clicking on someone to deal automatic all-hits (which is dealt with by this PR by having lower base damage), the "shotguns are superior at close range" understanding people have of video game shotguns really doesn't apply here. 

That's why I'm pressing to redefine narrow maint tunnels as what CQB combat should be defined around, rather than arbitrary "2 tiles away vs 4 tiles away" which doesn't matter nearly as much in ss13

**Comment by optimumtact** [comment]
>Most gun combat happens further than 3 tiles away

most combat doesn't

**Comment by Son-of-Space** [comment]
Do you know why I use the shotgun?

Because they added back even cooler buckshot

and it (only sometimes) misses.

**Comment by CRITAWAKETS** [comment]
buckshot honestly needed to go, pretty much all combat is done in close to melee range (otherwise its really easy to dodge projectiles + lag can mess up your shots)

Also, is there anything that makes this not better than a melee weapon in every way besides ammo consumption?

**Comment by Ryll-Ryll** [comment]
Well it's ammo consumption that you need to use cargo's shuttle to refill, which is markedly different than being printable

**Comment by GuillaumePrata** [comment]
When I heard the name Voidshot I expected the new buckshot to have an unique effect.
Standard Buckshots are always at this weird design space where their only use is to quickly lethal someone at close range, if they don't kill fast enough they don't get used, if they kill fast enough people will do whatever they can to get their hands into it.

People would make the ghetto buckshot every shift instead of using lasers, giving them access to a better version will likely remove the other more interesting options out of the officer's hands.

**Comment by zxaber** [comment]
Lesser damage (and thus inertia) per pellet, and less loss of velocity from air resistance hint at smaller pellet sizes. Why not just call it birdshot?

Market it ICly as anti space-carp.

**Comment by MMMiracles** [comment]
> Lesser damage (and thus inertia) per pellet, and less loss of velocity from air resistance hint at smaller pellet sizes. Why not just call it birdshot?
> 
> Market it ICly as anti space-carp.

buckshot
birdshot
carpshot

the only logical step in pellet-based technology

**Comment by Ryll-Ryll** [comment]
This has less pellets, not more

**Comment by Pandarsenic** [comment]
While I'm tentatively VERY excited about voidshot, I'm worried about the wounding power against solo antags who have very minimal ability to deal with their own wounds or very limited time to do it (Wizards, Nuke Ops, Ninjas, etc.)

If you don't mind, what's the average damage and shells-to-wound of each type of shot, vs. unarmored targets and vs. nuke op basic suit, at...
- point blank
- 3 tiles (open hallway and 1-wide maintenance hallway)
- 6 tiles (as above)
- 9 tiles (as above)

**Comment by Iatots** [comment]
Why give the standard a special name and a special the normal name?

**Comment by Coastfront** [comment]
the issue with re-adding buckshot, as oranges pointed out, basically all combat happens near point blank range.
The only 2 long range guns that are actually used at a long distance (farther than "i can melee you within 2 seconds of you shooting range") are a sniper rifle with a scope to see extremely far, and the beam rifle.
This is partly because of the fact that without a scope, any long range combat is you blind firing at someone, because people cant see far, and as a result of not seeing far, spaces where crew manuever tend to APPROPRIATELY be tight.

This means, any gun built to do better at close range combat than long range, will always out perform every other crew sided gun, because it will just be good at every reasonable circumstance you'll use it in.


dear fucking god i didnt think the day would come i agree with oranges on something

**Comment by Striders13** [comment]
Why can't we just make a gun good, why can't shotguns simply be a good choice? Shotguns should feel and be powerful because they're shotguns.
They SHOULD two shot anyone point blank, because if you're stupid enough to get close to a guy with a big gun instead of running away and spamming disabler at him you can eat shit.
Remember that it's impossible to hide shotgun in your backpack (unless you use saw which I think makes it a bad gun?), so you can plan your approach to shotgun users accordingly.

**Comment by Valorium-RB** [comment]
> > Why can't we just make a gun good, why can't shotguns simply be a good choice? Shotguns should feel and be powerful because they're shotguns. They SHOULD two shot anyone point blank, because if you're stupid enough to get close to a guy with a big gun instead of running away and spamming disabler at him you can eat shit. Remember that it's impossible to hide shotgun in your backpack (unless you use saw which I think makes it a bad gun?), so you can plan your approach to shotgun users accordingly.
> 
> The issue is the fact that almost all combat, becomes close range. Sure, open hallways you can blindfire at them from off screen, In maint or most departments, there are corners, with short hallways. Which is a cqc gauntlet. Shotguns are already extremely powerful because 2 hit stunning someone faster than a stun baton means you just kill them anyway. making it so shotguns kill someone faster than having to stun and then put them down puts them back at "arguably the best weapon that non nukies can get" territory. also iirc sawing off a shotgun has 0 debuffs to its 3-4 tile lethality.

I'm fairly certain that "7 round revolver that crits in two hits at any range without having to so much as look at cargo and fits in bags without modification" is superior to "7 round shotgun that only crits in two at very short range (regardless of how often that is), requires either cargo access or repeated B&E to feed and requires procuring a circular saw (which is harder with the recent changes) to make fit in bags". The significant controlling factor here is that a validhunter can't just steal a riot shotgun and an autolathe board, camp out in maint and print out five boxes of buckshot anymore. Breaking open a ton of crates just for ammo makes it impractical.

**Comment by Ryll-Ryll** [comment]
> While I'm tentatively VERY excited about voidshot, I'm worried about the wounding power against solo antags who have very minimal ability to deal with their own wounds or very limited time to do it (Wizards, Nuke Ops, Ninjas, etc.)
> 
> If you don't mind, what's the average damage and shells-to-wound of each type of shot, vs. unarmored targets and vs. nuke op basic suit, at...
> 
> * point blank
> * 3 tiles (open hallway and 1-wide maintenance hallway)
> * 6 tiles (as above)
> * 9 tiles (as above)

Wounding math is tricky and has a lot of factors at play, especially with shotguns since there's also the question of how many pellets actually hit, [but I took a video of shooting some targets with the values I have right now (top shotgun has voidshot, bottom has buckshot) to give an idea of what it'd look like](https://streamable.com/6t5qfn)

Voidshot has 5 pellets that do 8 damage each, losing .3 damage per pellet per tile, with bullet armor being doubly effective against it (the 30% on a sec vest jumps to 60% for example). The wounding power is far greater on targets who have no armor, especially because of the high bare_wound_bonus value that only applies when the victim has no armor. Against armored opponents (wizards, nukies, and ninjas all come with robust armor) it's far less effective and needs multiple direct hits to really shred them, though I might reduce the wounding against armor a bit more. Buckshot, on the other hand, has 6 pellets that do 9 damage now, and is far less affected by armor, though it remains basically only in the hands of syndies.

People talking about 2-hit crits here are misinformed- Voidshot does 40 damage max at point blank, not accounting for any armor, the .75x damage multiplier for limbs, etc that would further lower it.

**Comment by Redrover1760** [comment]
> While I'm tentatively VERY excited about voidshot, I'm worried about the wounding power against solo antags who have very minimal ability to deal with their own wounds or very limited time to do it (Wizards, Nuke Ops, Ninjas, etc.)
> 
> If you don't mind, what's the average damage and shells-to-wound of each type of shot, vs. unarmored targets and vs. nuke op basic suit, at...
> 
> * point blank
> * 3 tiles (open hallway and 1-wide maintenance hallway)
> * 6 tiles (as above)
> * 9 tiles (as above)

Nuke ops have like 6 bullet armor, and its doubled vs voidshot, so therefore they basically don't take shotgun damage.

**Comment by Ryll-Ryll** [comment]
oh lol i don't know how new nuke suits work, i just spawned them in that outfit like i used to. let me try again

**Comment by Ryll-Ryll** [comment]
#67331 has balance implications for this so I wanna wait for that to get merged first

**Comment by Farquaar** [comment]
> Do you know why I use the shotgun?
> 
Because it doesn't miss?

**Comment by Redrover1760** [comment]
> oh lol i don't know how new nuke suits work, i just spawned them in that outfit like i used to. let me try again
> 
> [new vid of armored nukies](https://streamable.com/j3jig9)

Yeah no that sounds about right (how tf is it still wounding despite doing 0 damage tho lol)

**Comment by Son-of-Space** [comment]
I've thought about this a lot and I have two thoughts here:
1. Voidshot is cool but buckshot (even with a skin) isn't really interesting. It probably won't be good for the game and will just simplify combat again.
2. I wish we would do something to address the other types of shot in the game which are so inconvenient to get that nobody does it. I think the ammo crafting piece is something that would be really cool and would enable us to actually explore the balance implications of that ammo in a more interesting way (laser scatter, meteor, frag, etc.) Perhaps we could do a little bit of both where science unlocks the ability to purchase parts and then they can be ordered through cargo? I think having the parts immediately at the lathe would be a little bit too far. The other piece to that is that crafting the ammo is very clunky atm.

The TLDR here is that more ammo variation both with energy and ballistic is where I think we can add the most value to the game and that this ammo type is a little bit generic and is effectively just adding back something that wasn't really great for the game without propping up other, existing areas of the system that we're lacking in.

**Comment by Ryll-Ryll** [comment]
> I've thought about this a lot and I have two thoughts here:
> 
> 1. Voidshot is cool but buckshot (even with a skin) isn't really interesting. It probably won't be good for the game and will just simplify combat again.
> 2. I wish we would do something to address the other types of shot in the game which are so inconvenient to get that nobody does it. I think the ammo crafting piece is something that would be really cool and would enable us to actually explore the balance implications of that ammo in a more interesting way (laser scatter, meteor, frag, etc.) Perhaps we could do a little bit of both where science unlocks the ability to purchase parts and then they can be ordered through cargo? I think having the parts immediately at the lathe would be a little bit too far. The other piece to that is that crafting the ammo is very clunky atm.
> 
> The TLDR here is that more ammo variation both with energy and ballistic is where I think we can add the most value to the game and that this ammo type is a little bit generic and is effectively just adding back something that wasn't really great for the game without propping up other, existing areas of the system that we're lacking in.

it's different because it wounds instead of kills, and i didn't need 7 paragraphs to get that across

**Comment by theta060** [comment]
> buckshot honestly needed to go, pretty much all combat is done in close to melee range (otherwise its really easy to dodge projectiles + lag can mess up your shots)
> 
> Also, is there anything that makes this not better than a melee weapon in every way besides ammo consumption?

a gun is always better then melee because shitty stun based combat
